CC/src/DistanceComputationTools.cpp

```diff
@@ -1177,23 +1173,25 @@ void cloudMeshDistCellFunc_MT(const DgmOctree::IndexAndCode& desc)
 	size_t trianglesToTestCapacity = 0;
 
 	//bit mask for efficient comparisons
-	QBitArray* bitArray = nullptr;
+	std::vector<bool>* bitArray = nullptr;
 	if (s_useBitArrays_MT)
 	{
-		s_currentBitMaskMutex.lock();
 		if (s_bitArrayPool_MT.empty())
 		{
-			bitArray = new QBitArray();
+			bitArray = new std::vector<bool>;
 			bitArray->resize(s_intersection_MT->mesh->size());
```

Comment:
This can just be:
```
bitArray = new std::vector<bool>(s_intersection_MT->mesh->size());
```
It default initializes to `false`, so the `std::fill` can be moved into the "else". There might be a better way to do that as well.